### ISSUE TITLE
MA0003 also reports for params

### ISSUE THREAD

**Issue by Piedone** [meziantou/Meziantou.Analyzer#276]
**Context**
Rule identifier: _MA0003_ (v1.0.681)
Target Framework: _.NET Core 3.1_
Language version: _9.0_

**Describe the bug**
[MA0003](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0003.md) also reports if the target parameter is a `params` one.

**To Reproduce**

This (not too sensical) `string.Format()` will result in an MA0003 violation despite the `public static string Format(string format, params object?[] args)` overload being used.

````c#
var myString = string.Format("Hi {0}, {1}, {2}, {3}.", null, null, null, null);
````

Note that while I used `string.Format()` in the example, this is an issues in cases where it actually makes sense to pass a lot of e.g. `null`s, just this is the simplest demonstration.

**Comment by meziantou** [comment]
Thanks for the report. It should be fixed in version 1.0.682 ([CI](https://github.com/meziantou/Meziantou.Analyzer/actions/runs/1631744809))

**Comment by Piedone** [comment]
Awesome, thanks for the quick fix. First download :).

However, it doesn't seem to be perfect yet. It fixes the example I mentioned but not these:
- https://github.com/Lombiq/Testing-Toolbox/blob/dev/Helpers/UserManagerMockHelpers.cs#L19
- https://github.com/Lombiq/Testing-Toolbox/blob/dev/Helpers/UserManagerMockHelpers.cs#L35

![image](https://user-images.githubusercontent.com/1976647/147605662-7dffd671-57cc-48a4-8b67-fa6c10165507.png)

**Comment by Piedone** [comment]
Hmm, interesting, this shows violations for me locally in Visual Studio, and during build as well (even after a `git clean`, but not in CI. I don't understand.

**Comment by meziantou** [comment]
I cannot tell about the CI issue. But I think I've found why it doesn't work with `new Mock`. In fact, the analyzer didn't support constructors correctly. Should be fixed by https://github.com/meziantou/Meziantou.Analyzer/pull/279

**Comment by meziantou** [comment]
Let me know if the version 1.0.683 works better :) (https://github.com/meziantou/Meziantou.Analyzer/actions/runs/1631915208)

**Comment by Piedone** [comment]
Thanks again for the quick fix. This works now!
